### PR TITLE
[IRGen] Query the VFS for legacy type layouts.

### PR DESCRIPTION
**Pull request by DougGregor**
All file I/O performed by the frontend should go through the VFS.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@benlangmuir am I holding it right?

**Review by benlangmuir — COMMENTED** [review]
LGTM

File: lib/IRGen/GenType.cpp [hunk 1]
```diff
@@ -1209,6 +1210,8 @@ TypeConverter::TypeConverter(IRGenModule &IGM)
   llvm::SmallString<128> defaultPath;
 
   StringRef path = IGM.IRGen.Opts.ReadLegacyTypeInfoPath;
+  auto fs =
+      IGM.getSwiftModule()->getASTContext().SourceMgr.getFileSystem();
```
Comment (jrose-apple) [inline lib/IRGen/GenType.cpp]: `IGM.Context.SourceMgr`

Comment (DougGregor) [reply lib/IRGen/GenType.cpp]: That's simpler, thanks!

**Pull request merged**